src/ray/pubsub/publisher.h

```diff
@@ -108,7 +127,7 @@ class Subscriber {
   bool PublishIfPossible(bool force = false);
 
   /// Testing only. Return true if there's no metadata remained in the private attribute.
-  bool AssertNoLeak() const;
+  bool CheckNoLeaks() const;
```

Comment:
Move to private if for testing.

Reply:
Will use better gtest solution or tons of FRIEND_TEST instead.

Reply:
For some classes like internal classes, this is necessary to be public, and I think it is fine for them because they are not meant to be re-used. But for the top level classes like Publisher and Subscriber, I will use friend test